### PR TITLE
Rithm add prop types

### PR DESCRIPTION
**Pull request by japamat0**
Issue [[#41](https://github.com/desmosinc/gifsmos/issues/41)]

I added prop types and default props to the following components:

App.js
Burst.js
ErrorToast.js
Frame.js
Preview.js
Settings.js
Sidebar.js
SidebarButton.js
SidebarButtonWithBadge.js

Default props for connected components have been set to match initial state from the store, with there mapDispatchToProps functions have been set as simple arrow functions, i.e. () => {}. For non connected components, the default props have been set to match the properties applied where those components are rendered, i.e. where <Frame /> is rendered within <Preview />

thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,9 @@
     "husky": "^1.3.1",
     "jest-dom": "^3.0.0",
     "prettier": "1.15.3",
-    "pretty-quick": "^1.11.1",
+    "pretty-quick": "^1.8.0",
+    "prop-types": "^15.7.2",
+    "react-testing-library": "^5.4.2",
```
Comment (misscoded) [inline package.json]: Remove

File: src/components/ErrorToast.js [hunk 2]
```diff
@@ -10,6 +11,10 @@ const ErrorToast = ({ message }) => {
   );
 };
 
+ErrorToast.propTypes = {
```
Comment (misscoded) [inline src/components/ErrorToast.js]: Be consistent -- propTypes before defaultProps or vice versa, not one or the other.

File: src/components/SidebarButton.test.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ afterEach(cleanup);
 
 describe('<SidebarButton/>', () => {
   it('renders without crashing', () => {
-    global.renderWithRedux(<SidebarButton />);
+    global.renderWithRedux(<SidebarButton icon="icon" />);
```
Comment (misscoded) [inline src/components/SidebarButton.test.js]: What does this addition do?

**Review by misscoded — CHANGES_REQUESTED** [review]
See comments.

Also, remove the testing stuff, unless there's a pressing reason to keep it.

File: src/components/SidebarButton.test.js (hunk 3 above)
Comment (japamat0) [reply src/components/SidebarButton.test.js]: removed, no longer needed with default props added

File: src/components/ErrorToast.js (hunk 2 above)
Comment (japamat0) [reply src/components/ErrorToast.js]: Fixed

File: package.json (hunk 1 above)
Comment (japamat0) [reply package.json]: Removed

**Review by misscoded — APPROVED** [review]
:shipit:

**Pull request merged**